### PR TITLE
Tests for bidders not satisfying the json-schema

### PR DESCRIPTION
**Pull request by mtuchkova**
Tests for #964

### CODE REVIEW

File: src/test/groovy/org/prebid/server/functional/BidderParamsSpec.groovy [hunk 1]
```diff
@@ -273,4 +277,55 @@ class BidderParamsSpec extends BaseSpec {
...
+        def response = defaultPbsService.sendAuctionRequest(bidRequest)
+
+        then: "Response should not contain errors"
+        assert !response.ext?.errors
+
+        and: "Response should contain warnings"
+        assert response.ext?.warnings[ErrorType.PREBID]*.message ==
+                ["WARNING: request.imp[0].ext.prebid.generic was dropped with a reason: " +
+                         "request.imp[0].ext.prebid.bidder.generic failed validation",
+                 "WARNING: request.imp[0].ext must contain at least one valid bidder"]
```
Comment (rpanchyk) [inline src/test/groovy/org/prebid/server/functional/BidderParamsSpec.groovy:298]: i'd rather recommend to test the case when `exampleProperty` given as string.

Comment (rpanchyk) [reply src/test/groovy/org/prebid/server/functional/BidderParamsSpec.groovy:298]: sorry, misunderstand. You're actually did in right way.

**Review by rpanchyk — DISMISSED** [review]

File: src/test/resources/org/prebid/server/functional/db_schema.sql [hunk 2]
```diff
@@ -42,6 +42,6 @@ CREATE TABLE stored_imps
 CREATE TABLE stored_responses
 (
     id     SERIAL PRIMARY KEY,
-    uuid   varchar(40) NOT NULL,
-    config varchar(1024)
+    resid   varchar(40) NOT NULL,
```
Comment (Net-burst) [inline src/test/resources/org/prebid/server/functional/db_schema.sql:45]: Why change that?

File: src/test/groovy/org/prebid/server/functional/model/request/auction/ImpExtPrebid.groovy [hunk 3]
```diff
@@ -6,6 +6,7 @@ import groovy.transform.ToString
 class ImpExtPrebid {
 
     Bidder bidder
+    StoredAuctionResponse storedAuctionResponse
```
Comment (Net-burst) [inline src/test/groovy/org/prebid/server/functional/model/request/auction/ImpExtPrebid.groovy:9]: You'll need to lowercase that for JSON processing.

File: src/test/groovy/org/prebid/server/functional/BidderParamsSpec.groovy [hunk 4]
```diff
@@ -273,4 +277,55 @@ class BidderParamsSpec extends BaseSpec {
...
+
+        when: "PBS processes auction request"
+        def response = defaultPbsService.sendAuctionRequest(bidRequest)
+
+        then: "Response should not contain errors"
+        assert !response.ext?.errors
+
+        and: "Response should contain warnings"
+        assert response.ext?.warnings[ErrorType.PREBID]*.message ==
+                ["WARNING: request.imp[0].ext.prebid.generic was dropped with a reason: " +
```
Comment (Net-burst) [inline src/test/groovy/org/prebid/server/functional/BidderParamsSpec.groovy:296]: Wording looks wrong. What exactly was dropped?

File: src/test/groovy/org/prebid/server/functional/BidderParamsSpec.groovy [hunk 5]
```diff
@@ -273,4 +277,55 @@ class BidderParamsSpec extends BaseSpec {
...
+            imp[0].ext.prebid.bidder.generic.exampleProperty = PBSUtils.randomNumber
+        }
+
+        when: "PBS processes auction request"
+        def response = defaultPbsService.sendAuctionRequest(bidRequest)
+
+        then: "Response should not contain errors"
+        assert !response.ext?.errors
+
+        and: "Response should contain warnings"
```
Comment (Net-burst) [inline src/test/groovy/org/prebid/server/functional/BidderParamsSpec.groovy:294]: Please be more exact. Bidder should be dropped.

File: src/test/groovy/org/prebid/server/functional/BidderParamsSpec.groovy [hunk 6]
```diff
@@ -273,4 +277,55 @@ class BidderParamsSpec extends BaseSpec {
...
+
+        and: "PBS should not call bidder"
+        assert bidder.getRequestCount(bidRequest.id) == 0
+
+        and: "seatbid should be empty"
+        assert response.seatbid.isEmpty()
+    }
+
+    @PendingFeature
+    def "PBS should not fail auction with storedAuctionResponse when request bidder params doesn't satisfy json-schema"() {
```
Comment (Net-burst) [inline src/test/groovy/org/prebid/server/functional/BidderParamsSpec.groovy:308]: Kinda pointless test. Stored auction response is for debugging purposes. Better check that the same logic is valid for AMP and stored requests.

**Review by Net-burst — CHANGES_REQUESTED** [review]

**Review by Net-burst — APPROVED** [review]
LGTM

**Review by rpanchyk — APPROVED** [review]

**Pull request merged**